webui/react/src/components/TasksTable.tsx

```diff
@@ -0,0 +1,100 @@
+import { Table } from 'antd';
+import { ColumnsType } from 'antd/lib/table';
+import React from 'react';
+import TimeAgo from 'timeago-react';
+
+import Avatar from 'components/Avatar';
+import Badge from 'components/Badge';
+import linkCss from 'components/Link.module.scss';
+import { CommonProps, Task } from 'types';
+import { alphanumericSorter, stateSorter, stringTimeSorter } from 'utils/data';
+import { canBeOpened } from 'utils/task';
+
+import { BadgeType } from './Badge';
+import Icon from './Icon';
+import { handleClick } from './Link';
+import TaskActionDropdown from './TaskActionDropdown';
+import css from './TasksTable.module.scss';
+
+interface Props extends CommonProps {
+  tasks: Task[];
+}
+
+  type Renderer<T> = (text: string, record: T, index: number) => React.ReactNode
+
+const typeRenderer: Renderer<Task> = (_, record) =>
+  (<Icon name={record.type.toLowerCase()} />);
+const startTimeRenderer: Renderer<Task> = (_, record) =>
+  (
+    <span title={record.startTime}>
+      <TimeAgo datetime={record.startTime} />
+    </span>
+  );
```

Comment:
nit: same layout as `stateRenderer`?
```
const startTimeRenderer: Renderer<Task> = (_, record) => (
  <span title={record.startTime}>
    <TimeAgo datetime={record.startTime} />
  </span>
);
```

Reply:
thought: what do you think about naming the `_` parameter? using the `_` or `_unused` prefix to indicate that it's not currently used in the function, but still informing that the parameter holds X data incase we do want to use in.

Reply:
I think `_` is a special case otherwise the linter would complain. if not, that sounds good.

Reply:
it doesn't complain but thinking about it more I see `_` as a convention in places to show ignored variables `_xyz` doesn't play that role :thinking: either way works though.

Reply:
`_` is easier to ignore also when reading while reading the code. good or bad